proto/internal/temporal/server/api/enums/v1/task.proto

```diff
@@ -49,18 +51,22 @@ enum TaskType {
     TASK_TYPE_REPLICATION_SYNC_ACTIVITY = 2;
     TASK_TYPE_TRANSFER_WORKFLOW_TASK = 3;
     TASK_TYPE_TRANSFER_ACTIVITY_TASK = 4;
-    TASK_TYPE_TRANSFER_CLOSE_EXECUTION = 5;
+    TASK_TYPE_TRANSFER_CLOSE_EXECUTION = 5 [ deprecated = true ];
     TASK_TYPE_TRANSFER_CANCEL_EXECUTION = 6;
     TASK_TYPE_TRANSFER_START_CHILD_EXECUTION = 7;
     TASK_TYPE_TRANSFER_SIGNAL_EXECUTION = 8;
-    TASK_TYPE_TRANSFER_RECORD_WORKFLOW_STARTED = 9;
+    TASK_TYPE_TRANSFER_RECORD_WORKFLOW_STARTED = 9 [ deprecated = true ];
     TASK_TYPE_TRANSFER_RESET_WORKFLOW = 10;
-    TASK_TYPE_TRANSFER_UPSERT_WORKFLOW_SEARCH_ATTRIBUTES = 11;
+    TASK_TYPE_TRANSFER_UPSERT_WORKFLOW_SEARCH_ATTRIBUTES = 11 [ deprecated = true ];
     TASK_TYPE_WORKFLOW_TASK_TIMEOUT = 12;
     TASK_TYPE_ACTIVITY_TIMEOUT = 13;
     TASK_TYPE_USER_TIMER = 14;
     TASK_TYPE_WORKFLOW_RUN_TIMEOUT = 15;
     TASK_TYPE_DELETE_HISTORY_EVENT = 16;
```

Comment:
TASK_TYPE_DELETE_HISTORY_EVENT -> TASK_TYPE_VISIBILITY_DELETE_EXECUTION?

Reply:
If I get it right, `TASK_TYPE_DELETE_HISTORY_EVENT` is a timer task which is fired when it is time to delete workflow. This is operation by itself is multi step operation and one step is to clear visibility.